### PR TITLE
Create _workspaces.md

### PR DESCRIPTION
**Pull request by HannaTrotsenko**
Add Workspace endpoint to the api-docs

### CODE REVIEW

File: source/includes/_workspaces.md [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+
+- Workspace Admin - this role gives the holder all the rights within the Workspace unit
+- Integrator - this role is usually granted for designing the integration process by building the flows, etc
+- Guest - this role gives view-only rights to examine the work done by Integrators
+
+Each role is limited to the given Workspace only. The same user in the platform can have different roles in different Workspaces
+
+Request / Role| Tenant Admin | Workspace Admin | Integrator | Guest
+---------- | :---------:| :------------:| :-----------:| :----------:
+Create a Workspace|- |X|X|X|
```
Comment (yarikos) [inline source/includes/_workspaces.md]: This operation belongs to Contract level, so I'd suggest removing this row from the table.

File: source/includes/_workspaces.md [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+All members of a Workspace would have their access level or roles.
+Currently, there are only 3 roles:
+
+- Workspace Admin - this role gives the holder all the rights within the Workspace unit
+- Integrator - this role is usually granted for designing the integration process by building the flows, etc
+- Guest - this role gives view-only rights to examine the work done by Integrators
+
+Each role is limited to the given Workspace only. The same user in the platform can have different roles in different Workspaces
+
+Request / Role| Tenant Admin | Workspace Admin | Integrator | Guest
```
Comment (yarikos) [inline source/includes/_workspaces.md]: `Tenant Admin` has no any special permissions regarding Workspaces, so I'd suggest removing the column at all.

File: source/includes/_workspaces.md [hunk 3]
```diff
@@ -0,0 +1,724 @@
...
+  "meta":{
+
+  },
+  "links":{
+    "self":"/v2/workspaces"
+  }
+}
+```
+
+This endpoint returns a Workspaces objects which belong to the given User.
```
Comment (yarikos) [inline source/includes/_workspaces.md]: returns a list of Workspaces

File: source/includes/_workspaces.md [hunk 4]
```diff
@@ -0,0 +1,724 @@
...
+  -u {EMAIL}:{APIKEY}
+```
+
+> Example Response:
+
+```http
+HTTP/1.1 204 No Content
+```
+
+This endpoint will delete an Workspace along with the following items that were inside the Workspace:
```
Comment (yarikos) [inline source/includes/_workspaces.md]: ...will delete the Workspace...

File: source/includes/_workspaces.md [hunk 5]
```diff
@@ -0,0 +1,724 @@
...
+* Flow's DynamicSelectModel
+* Flow'sExecStat
+* Flow's ExecutionResult
+* Flow's MarathonEvent
+* Flow's RequestBin
+* Flow's TaskHooksData
+* Flow's TaskStats
+* Flow's TaskStatError
+* Flow's TaskVersion
+* Teams
```
Comment (yarikos) [inline source/includes/_workspaces.md]: Remove Teams, Repos and RepoBuilds – they belong to Contract.

File: source/includes/_workspaces.md [hunk 6]
```diff
@@ -0,0 +1,724 @@
...
+* Repos
+* RepoBuilds
+* User accounts who were only the members of the deleted Organization
```
Comment (yarikos) [inline source/includes/_workspaces.md]: Remove this row

File: source/includes/_workspaces.md [hunk 7]
```diff
@@ -0,0 +1,724 @@
...
+???
+
+### HTTP Request
+`DELETE https://api.elastic.io/v2/workspaces/{WORKSPACE_ID} \`
+
+#### Authorization
+This request is authorized for organization members with role `Workspace Admin`.
```
Comment (yarikos) [inline source/includes/_workspaces.md]: Pls add a note, that Workspace is not deleted immediately (you can use Contract delete for example)

File: source/includes/_workspaces.md [hunk 8]
```diff
@@ -0,0 +1,724 @@
...
+
+???
+* Accounts (Credentials)
+* Agents
+* DataSamples
+* InviteTokens
+* Lookups
+* Flow's DynamicMetadata
+* Flow's DynamicSelectModel
+* Flow'sExecStat
```
Comment (yarikos) [inline source/includes/_workspaces.md]: Missing space

File: source/includes/_workspaces.md [hunk 9]
```diff
@@ -0,0 +1,724 @@
...
+> Example Response:
+
+```http
+HTTP/1.1 204 No Content
+```
+
+This endpoint will delete an Workspace along with the following items that were inside the Workspace:
+
+???
+* Accounts (Credentials)
```
Comment (yarikos) [inline source/includes/_workspaces.md]: Pls remove `Accounts`, leave only `Credentials` (and pls do the same for Contracts)

File: source/includes/_workspaces.md (hunk 1 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: we decided to leave it here

File: source/includes/_workspaces.md (hunk 2 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 3 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md [hunk 10]
```diff
@@ -0,0 +1,724 @@
...
+### HTTP Request
+
+`GET https://api.elastic.io/v2/workspaces`
+
+#### Authorization
+
+User has to be a member of the Workspace.
+
+
+### URL Parameters
```
[the remark this answers is not in the document]
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md [hunk 11]
```diff
@@ -0,0 +1,724 @@
...
+> Example Response:
+
+```shell
+HTTP/1.1 204 No Content
+```
+
+Remove a membership of the User in the Workspace.
+Ownership of those user's associated data will be transferred to Admin User performing this operation:
+
+* ???
```
[the remark this answers is not in the document]
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: I don't know exactly what data will be transferred

File: source/includes/_workspaces.md (hunk 4 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 5 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 6 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 7 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 8 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

File: source/includes/_workspaces.md (hunk 9 above)
Comment (HannaTrotsenko) [reply source/includes/_workspaces.md]: done

**Pull request merged**